node/ChainwebNode.hs

```diff
@@ -427,6 +429,43 @@ mkTelemetryLogger
 mkTelemetryLogger mgr = configureHandler
     $ withJsonHandleBackend @(JsonLog a) (sshow $ typeRep $ Proxy @a) mgr pkgInfoScopes
 
+-- -------------------------------------------------------------------------- --
+-- Kill Switch
+
+withKillSwitch
+    :: (LogLevel -> T.Text -> IO ())
+    -> Maybe UTCTime
+    -> IO a
+    -> IO a
+withKillSwitch _ Nothing inner = inner
+withKillSwitch lf (Just t) inner = race timer inner >>= \case
+    Left () -> error "kill switch thread termianted unexpectedly"
+    Right a -> return a
+  where
+    timer = do
+        runForever lf "KillSwitch" $ do
+            now <- getCurrentTime
+            when (now >= t) $ do
+                lf Error killMessage
+                error $ T.unpack killMessage
```

Comment:
`error` isn't an asynchronous exception, so it won't terminate the `runForever` loop.